Tasks/Src/PetWatchdogTask.c

```diff
@@ -0,0 +1,84 @@
+/*-----------------------------------------------------------
+ * PET WATCHDOG TASK
+ - Attempts to pet watchdog within appropriate time interval
+ -----------------------------------------------------------*/
+#include "BPS_Tasks.h"
+#include "WDog.h"
+#include <timers.h>
+
+
+/**
+ * @brief Watchdog Event Group Initialization
+ */
+void Init_WDogEventGroup() {
+    // Event Group init
+    xWDogEventGroup_handle = xEventGroupCreateStatic( &xWDogEventGroup );
+    configASSERT( xWDogEventGroup_handle );          // check if handle is set 
+    xEventGroupClearBits(xWDogEventGroup_handle,     /* The event group being updated. */
+                         0xFF );                /* The bits being cleared. */
+}
+
+
+/**
+ * @brief Callback function for Watchdog Window Timer.
+ *        Sets event group bit to signal Watchdog.
+ */
+void WDog_WindowCallback(TimerHandle_t xTimer) {
+    xEventGroupSetBits(xWDogEventGroup_handle, TIMER_DONE);
+}
+
+/*--------------------------------------------------------*/
+
+void Task_PETWDOG() {
+    // Initialize GPIO output for Pins 5 (LED), 6 (Task 1), 7 (Task 2)
+    GPIO_InitTypeDef gpio_config = {
+        .Mode = GPIO_MODE_OUTPUT_PP,
+        .Pull = GPIO_NOPULL,
+        .Pin = GPIO_PIN_5 | GPIO_PIN_6 | GPIO_PIN_7
+    };
```

Comment:
- If the task 1 and task 2 pins are just debugging instruments do you still need them to be around?
- Also, you may want to initialize the LED somewhere else that applies globally (init task) so that any task can use it as a debug tool/indicator. However if it's also just a temporary debug instrument you should remove it from the final commit